### DISCUSSION TITLE
WICG Contact Picker API

### DISCUSSION THREAD

**Discussion by elf-pavlik** [solid/data-interoperability-panel#112] [Ideas]
https://github.com/WICG/contact-api

> This is a proposal for adding a Contact Picker API to the Web. Contact pickers are frequently seen in native mobile applications for a variety of [use cases](#use-cases), and in various desktop applications such as e-mail clients and calendars.
>
> Contact information is among the most sensitive information on a device, which is why we're proposing a picker rather than a lower level API. This enables user agents to offer an experience that's consistent with platform expectations, and to highlight the exact information that is to be shared with the website.
>
> This proposal does not address the source of the contacts. Many operating systems offer functionality to access the user's contacts, and various browsers allow the user to log in which might provide access to contact information as well.

--------------

Possibly something to keep an eye on, especially how in browser solid apps could take advantage from it and what other apps would come useful to sync operating system contacts with solid storage.

**Comment by ericprud** [comment]
[WICG](https://www.w3.org/community/wicg/) is taking the opposite approach from LDP. Where WICG is focused on specialized APIs, LDP focused on REST over specialized containers and shapes (see [Martin Nally's presentation](https://pdfslide.net/documents/-ibm-software-group-ibm-corporation-linked-data-as-an-application-integration.html)).

That's not to say that the optimum isn't a mix, but in order to be consistent with the platform, we should *try* to interpret such APIs as shapes rather than functions. We'll get good architectural insight where that doesn't work, or does awkwardly.

**Comment by elf-pavlik** [comment]
This video from recent Chrome Dev Summit has little (2 min) demo of Contact Picker API: https://youtu.be/JKVZMqpiY7w?t=482

I like the part where script can choose which properties they want to get and in a picker user may unselect sharing any specific one. In the demo they unselected phone nr and just shared name and email.
